<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>712</width>
    <height>890</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>100</horstretch>
    <verstretch>100</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Deetect: a deep learning based image analysis tool</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="styleSheet">
       <string notr="true">QGroupBox {
    border: 1px solid gray;
    border-radius: 9px;
    margin-top: 0.5em;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
      </property>
      <property name="title">
       <string>Select Images or Directory</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QGroupBox" name="groupBox_5">
         <property name="title">
          <string>Input</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QPushButton" name="opendir_btn">
            <property name="styleSheet">
             <string notr="true"> border: 3px solid gray;
border-radius: 9px; 
border-color: rgb(228, 166, 33);</string>
            </property>
            <property name="text">
             <string>Select Directory</string>
            </property>
            <property name="icon">
             <iconset>
              <normaloff>../gui_qt5/images/open_folder.ico</normaloff>../gui_qt5/images/open_folder.ico</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="fileopen_btn">
            <property name="styleSheet">
             <string notr="true"> border: 3px solid gray;
border-radius: 9px; 
border-color: rgb(1, 215, 222);</string>
            </property>
            <property name="text">
             <string>Select Images</string>
            </property>
            <property name="icon">
             <iconset>
              <normaloff>../gui_qt5/images/open_tif.ico</normaloff>../gui_qt5/images/open_tif.ico</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>24</width>
              <height>24</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_7">
         <property name="title">
          <string>Output Directory</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLineEdit" name="output_lineEdit"/>
          </item>
          <item>
           <widget class="QPushButton" name="outputDir_btn">
            <property name="toolTip">
             <string>Browse for output folder</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset>
              <normaloff>../cellpose_pipeline/images/open_folder.ico</normaloff>../cellpose_pipeline/images/open_folder.ico</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="filenr_infolabel">
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Number of images to be processed: 0</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_6">
      <property name="styleSheet">
       <string notr="true">QGroupBox {
    border: 1px solid gray;
    border-radius: 9px;
    margin-top: 0.5em;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
      </property>
      <property name="title">
       <string>Z-projection (Maximum Intensity Projection)</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>5</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>5</number>
       </property>
       <property name="bottomMargin">
        <number>5</number>
       </property>
       <property name="spacing">
        <number>5</number>
       </property>
       <item row="0" column="0">
        <widget class="QFrame" name="frame">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="topMargin">
           <number>5</number>
          </property>
          <property name="bottomMargin">
           <number>5</number>
          </property>
          <item>
           <widget class="QRadioButton" name="noprojection_rb">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>no projection</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="stack2by2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>2 by 2</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="stack5by5">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>5 by 5</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="stackallscans_rb">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>complete projection (all slices)</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="styleSheet">
       <string notr="true">QGroupBox {
    border: 1px solid gray;
    border-radius: 9px;
    margin-top: 0.5em;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
      </property>
      <property name="title">
       <string>Analyses and Parameters</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="bottomMargin">
        <number>15</number>
       </property>
       <property name="verticalSpacing">
        <number>3</number>
       </property>
       <item row="7" column="0">
        <widget class="QCheckBox" name="elimination_cb">
         <property name="styleSheet">
          <string notr="true">color: blue</string>
         </property>
         <property name="text">
          <string>Invisible, Small and Big Cell Detection </string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="cellpose_cb">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string notr="true">color:  green

</string>
         </property>
         <property name="text">
          <string>Nuclei Detection (via cellpose)</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <property name="tristate">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QGroupBox" name="elimination_gbox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string notr="true">QGroupBox {
    border: 1px solid gray;
    border-radius: 9px;
    margin-top: 0.5em;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
	color: gray;
}</string>
         </property>
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <property name="topMargin">
           <number>5</number>
          </property>
          <property name="bottomMargin">
           <number>5</number>
          </property>
          <property name="verticalSpacing">
           <number>6</number>
          </property>
          <item row="4" column="1">
           <widget class="QLineEdit" name="area_tbox">
            <property name="text">
             <string>400</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="intens_tbox">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;between 0-255&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>16</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="styleSheet">
             <string notr="true">QLabel {background-color : black; color : rgb(0, 170, 255); }
</string>
            </property>
            <property name="text">
             <string>  Pixel intensity threshold  to ignore invisible cell:    </string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_4">
            <property name="styleSheet">
             <string notr="true">QLabel {  background-color : black;color : gray; }</string>
            </property>
            <property name="text">
             <string>  Cell area limit to split big cell:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_7">
            <property name="styleSheet">
             <string notr="true">QLabel {background-color : black; color : yellow; }</string>
            </property>
            <property name="text">
             <string>  Cell area limit to ignore small cell</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="small_area_tbox">
            <property name="text">
             <string>100</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QCheckBox" name="doublecount_cb">
         <property name="font">
          <font>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">color:  red;</string>
         </property>
         <property name="text">
          <string>Double Counted Cell Detection</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QGroupBox" name="doublecount_gbox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string notr="true">QGroupBox {
    border: 1px solid red;
    border-radius: 9px;
    margin-top: 0.5em;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
         </property>
         <property name="title">
          <string/>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="topMargin">
           <number>5</number>
          </property>
          <property name="bottomMargin">
           <number>5</number>
          </property>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="styleSheet">
             <string notr="true">color:red</string>
            </property>
            <property name="text">
             <string>Intersection percent of two cells to set double counting:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="intersect_tbox">
            <property name="text">
             <string>60</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>%</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QGroupBox" name="groupBox_4">
         <property name="styleSheet">
          <string notr="true">QGroupBox {
    border: 1px  solid green;
    border-radius: 9px;
    margin-top: 0.5em;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
	 color:  blue;
}</string>
         </property>
         <property name="title">
          <string/>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <property name="topMargin">
           <number>5</number>
          </property>
          <property name="bottomMargin">
           <number>5</number>
          </property>
          <item row="1" column="0">
           <widget class="QLabel" name="label_8">
            <property name="toolTip">
             <string>The default is flow_threshold=0.4. Increase this threshold if cellpose is not returning as many masks as you’d expect. Similarly, decrease this threshold if cellpose is returning too many ill-shaped masks.</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color:green</string>
            </property>
            <property name="text">
             <string>Pre-trained network:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label">
            <property name="toolTip">
             <string>The default is flow_threshold=0.4. Increase this threshold if cellpose is not returning as many masks as you’d expect. Similarly, decrease this threshold if cellpose is returning too many ill-shaped masks.</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color:green</string>
            </property>
            <property name="text">
             <string>Flow threshold:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="flow_tbox">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Increase this threshold if cellpose is not returning as many masks as you’d expect. Similarly, decrease this threshold if cellpose is returning too many ill-shaped masks. Default=0.4&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>0.6</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_2">
            <property name="toolTip">
             <string>The default is mask_threshold=0.0. Decrease this threshold if cellpose is not returning as many masks as you’d expect. Similarly, increase this threshold if cellpose is returning too masks particularly from dim areas.</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color:green</string>
            </property>
            <property name="text">
             <string>Mask threshold:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLineEdit" name="cellprob_tbox">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Decrease this threshold if cellpose is not returning as many masks as you’d expect. Similarly, increase this threshold if cellpose is returning too masks particularly from dim areas. Range (-6 to +6)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QFrame" name="frame_4">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <property name="lineWidth">
             <number>0</number>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QRadioButton" name="cyto_rb">
               <property name="text">
                <string>cyto</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="cyto2_rb">
               <property name="text">
                <string>cyto2</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="nuclei_rb">
               <property name="text">
                <string>nuclei</string>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="run_btn">
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="toolTip">
       <string>Run all selected analyses</string>
      </property>
      <property name="styleSheet">
       <string notr="true"> border: 3px solid gray;
border-radius: 9px; 
border-color: rgb(143, 240, 0);</string>
      </property>
      <property name="text">
       <string>Run Selected Analyses</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>../gui_qt5/images/run.ico</normaloff>../gui_qt5/images/run.ico</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>30</width>
        <height>30</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QProgressBar" name="progressBar">
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="status_infolbl">
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Processed picture:</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>712</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuRun_Stats">
    <property name="title">
     <string>Run Stats</string>
    </property>
    <addaction name="actionOpen_Analysis_File"/>
   </widget>
   <addaction name="menuRun_Stats"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen_Analysis_File">
   <property name="icon">
    <iconset>
     <normalon>images/open_folder.ico</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Open Analysis File</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>stack5by5</tabstop>
  <tabstop>cellpose_cb</tabstop>
  <tabstop>intens_tbox</tabstop>
  <tabstop>area_tbox</tabstop>
  <tabstop>doublecount_cb</tabstop>
  <tabstop>intersect_tbox</tabstop>
  <tabstop>run_btn</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
